src/public/media.ts

```diff
@@ -77,3 +81,233 @@ export function captureImage(callback: (error: SdkError, files: File[]) => void)
   const messageId = sendMessageRequestToParent('captureImage');
   GlobalVars.callbacks[messageId] = callback;
 }
+
+/**
+ * Media object returned by the select Media API
+ */
+export class Media extends File {
+  /**
+   * A preview of the file which is a lightweight representation.
+   * In case of images this will be a thumbnail in base64 encoding.
+   */
+  public preview: string;
+
+  /**
+   * Gets the media in chunks irrespecitve of size, these chunks are assembled and sent back to the webapp as file/blob
+   * @param callback returns blob of media
+   */
+  public getMedia(callback: (error: SdkError, blob: Blob) => void): void {
+    ensureInitialized(frameContexts.content, frameContexts.task);
+    let actionName = generateGUID();
+    let helper: MediaHelper = {
+      mediaMimeType: this.mimeType,
+      assembleAttachment: [],
+    };
+    const params = [actionName, this.content];
+    GlobalVars.getMediaHandler = callback;
+    this.content && callback && sendMessageRequestToParent('getMedia', params);
+    function handleGetMediaRequest(response: string): void {
+      if (GlobalVars.getMediaHandler) {
+        let mediaResult: MediaResult = JSON.parse(response);
+        if (mediaResult.error) {
+          GlobalVars.getMediaHandler(mediaResult.error, null);
+        } else {
+          if (mediaResult.mediaChunk) {
+            if (mediaResult.mediaChunk.chunkSequence <= 0) {
+              let file = createFile(helper.assembleAttachment, helper.mediaMimeType);
+              GlobalVars.getMediaHandler(mediaResult.error, file);
+            } else {
+              let assemble: AssembleAttachment = decodeAttachment(mediaResult.mediaChunk, helper.mediaMimeType);
+              helper.assembleAttachment.push(assemble);
+            }
+          } else {
+            GlobalVars.getMediaHandler(
+              { errorCode: ErrorCode.INTERNAL_ERROR, message: 'data receieved is null' },
+              null,
+            );
+          }
+        }
+      }
+    }
+
+    GlobalVars.handlers['getMedia' + actionName] = handleGetMediaRequest;
```

Comment:
Yes. We need to send the actionName to the platform in order to distinguish between 2 parallel requests. Here the callback comes back multiple times unlike the other APIs, hence we need a unique identifier. We cannot use the one returned by sendMessageRequestToParent because we need before that to be sent in its params.

Reply:
When will these be unregistered? I do not see the code that removes these.

Reply:
done